src/strategies/base.ts

```diff
@@ -254,30 +268,145 @@ export abstract class BaseStrategy implements Strategy {
+    if (existingPullRequest) {
+      this.logger.info(
+        `PR already exists for ${existingPullRequest.headBranchName}, checking if PR title edited to set custom version`
+      );
+      const existingPRTitleVersion = PullRequestTitle.parse(
+        existingPullRequest.title
+      )?.getVersion();
+      const hasCustomVersionLabel = existingPullRequest.labels.find(
+        label => label === DEFAULT_CUSTOM_VERSION_LABEL
+      );
+
+      if (!existingPRTitleVersion && hasCustomVersionLabel) {
+        // report problem to end user
+        this.github.commentOnIssue(
+          `
+## Invalid version number in PR title
+
+:rotating_light: This Pull Request has the \`${DEFAULT_CUSTOM_VERSION_LABEL}\` label but the version number cannot be found in the title. Instead the generated version \`${newVersion}\` will be used.
+
+If you want to set a custom version be sure to use the [semantic versioning format](https://devhints.io/semver).
+
+If you do not want to set a custom version and want  to get rid of this warning, remove the label \`${DEFAULT_CUSTOM_VERSION_LABEL}\` from this Pull Request.
+`,
+          existingPullRequest.number
+        );
+      } else if (!existingPRTitleVersion) {
+        // warn end user
+        this.github.commentOnIssue(
+          `
+## Invalid version number in PR title
+
+:warning: No version number can be found in the title. Did you want to set a custom version for this release?
+
+To set a custom version be sure to use the semver format, e.g \`1.2.3\`.
+`,
+          existingPullRequest.number
+        );
+      } else if (
+        existingPullRequest.labels.find(
+          label => label === DEFAULT_CUSTOM_VERSION_LABEL
+        )
+      ) {
+        // PR labeled as custom version, use version from the title
+        newVersion = existingPRTitleVersion;
+      } else {
+        // look at the manifest from release branch and compare against version from PR title
+        try {
+          const manifest =
+            (await this.github.getFileJson<Record<string, string>>(
+              manifestPath || DEFAULT_RELEASE_PLEASE_MANIFEST,
+              existingPullRequest.headBranchName
+            )) || {};
+          const componentVersion = manifest[component || '.'];
+          if (componentVersion !== existingPRTitleVersion?.toString()) {
+            // version from title has been edited, add custom version label, a comment, and use the title version
+            this.github.addIssueLabels(
+              [DEFAULT_CUSTOM_VERSION_LABEL],
+              existingPullRequest.number
+            );
+            this.github.commentOnIssue(
+              `
+## Release version edited manually
+
+The Pull Request version has been manually set to \`${existingPRTitleVersion}\` and will be used for the release.
+
+If you instead want to use the version number \`${newVersion}\` generated from conventional commits, just remove the label \`${DEFAULT_CUSTOM_VERSION_LABEL}\` from this Pull Request.
```

Comment:
So if the custom label is removed, we ignore the version in the title?

Reply:
Yes, if the custom label is removed that triggers the github workflow and we will rely on conventional commits to determine the version bump.

Reply:
How do you distinguish between "user sets version in title" and "user set version title yesterday and removes label today". Don't they come out to the same state?

Reply:
I'm not sure I understand your concerns here but I will try to answer.

1. The PR is created, the version is calculated as `0.0.3`.
1. The user sets the version on Monday by editing the PR title. Because the version in the title does not match the version in the manifest for the PR branch, the version is now set to `1.2.3` and a label `custom version` is added.
2. On Tuesday they remove the label, which triggers the github action due to the `pull_request unlabeled` event. Because the version in the title does match the version from the manifest for the PR branch and the PR does not have a `custom version` label, the version from the title isn't used.

That part of the code isn't the most trivial to follow, but you can see that we first set `newVersion` based on information from commits. And only after do we check if we should or not set use the version from the PR title:

https://github.com/stainless-api/release-please/blob/eced66d374253fbed098945e289a5ea9a5bbcc74/src/strategies/base.ts#L323-L333

Reply:
Ok, got it. You try to determine the user action by comparing the title to the manifest, together with the existence of the label. It's probably good enough, although it feels there are corner cases we might encounter.